src/odemis/gui/cont/acquisition.py

```diff
@@ -735,19 +743,57 @@ def _on_streams_change(self, streams):
         ]
         self._sort_streams(item_stream)
 
+        # update the zlevels dictionary with the added/removed stream, if required
+        if self._zStackActive.value:
+            self._on_zstack()
+
     @wxlimit_invocation(1)  # max 1/s
     def _update_acquisition_time(self):
         """
         Updates the estimated time
         required for acquisition
         """
-        acq_time = acqmng.estimateTime(self._acquiStreams.value)
-        acq_time = math.ceil(acq_time)  # round a bit pessimistic
-        # TODO take into account the time of acquiring the zlevels
-        # as well as the time required for moving the actuator
+        if not self._zStackActive.value: # if no zstack 
+            acq_time = acqmng.estimateTime(self._acquiStreams.value)
+            acq_time = math.ceil(acq_time)  # round a bit pessimistic
+        
+        else: # if zstack 
+            acq_time = 0
+            nr_FM_streams = 0
+            for s in self._acquiStreams.value:
+                acq_time += s.estimateAcquisitionTime() * len(self._zlevels[s])  
+                acq_time = math.ceil(acq_time)              
+                if isinstance(s, FluoStream):
+                    nr_FM_streams += 1
+
+            z_pos = next((l for l in self._zlevels.values() if len(l) > 1), None)
+            if z_pos:
+                zstep = abs(z_pos[0]-z_pos[1])
```

Comment:
A comment about what the code is trying to do would be nice. Specially that this estimation is for FM streams only.